### PR TITLE
Documentation guide

### LINKED ISSUE #778: Add documentation building to the Developer's Guide
In order to encourage contributors to update the documentations it would be good to have a short tutorial on docs building:

- install sphinx
- install dependencies
- build HTML

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
## Proposed changes

Fixes #778 by adding a guide to building the documentation in the docs.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
@KulaginVladimir do you think there's anything I missed here?

File: docs/source/devguide/index.rst [hunk 1]
```diff
@@ -110,3 +110,60 @@ Implementing a new feature
...
+
+To build the documentation locally, you can use the following command:
+
+.. code-block:: bash
+
+   cd docs/source
+   make html
+
+This will generate the documentation in the `docs/source/_build/html` folder.
+You can then open the `index.html` file in your browser to see the documentation.
```
Comment (KulaginVladimir) [inline docs/source/devguide/index.rst:134]: Maybe, `make clean` is also worth mentioning?

Comment (RemDelaporteMathurin) [reply docs/source/devguide/index.rst:134]: Hm maybe, what does it do? I usually use the `sphinx-build` command

Comment (KulaginVladimir) [reply docs/source/devguide/index.rst:135]: The list of available arguments for `make`:
```
Please use `make target' where target is one of
  html        to make standalone HTML files
  dirhtml     to make HTML files named index.html in directories
  singlehtml  to make a single large HTML file
  pickle      to make pickle files
  json        to make JSON files
  htmlhelp    to make HTML files and an HTML help project
  qthelp      to make HTML files and a qthelp project
  devhelp     to make HTML files and a Devhelp project
  epub        to make an epub
  latex       to make LaTeX files, you can set PAPER=a4 or PAPER=letter
  latexpdf    to make LaTeX and PDF files (default pdflatex)
  latexpdfja  to make LaTeX files and run them through platex/dvipdfmx
  text        to make text files
  man         to make manual pages
  texinfo     to make Texinfo files
  info        to make Texinfo files and run them through makeinfo
  gettext     to make PO message catalogs
  changes     to make an overview of all changed/added/deprecated items
  xml         to make Docutils-native XML files
  pseudoxml   to make pseudoxml-XML files for display purposes
  linkcheck   to check all external links for integrity
  doctest     to run all doctests embedded in the documentation (if enabled)
  coverage    to run coverage check of the documentation (if enabled)
  clean       to remove everything in the build directory
```

Comment (RemDelaporteMathurin) [reply docs/source/devguide/index.rst:135]: Oh yes I remembered I used `make doctest` for another project. Probably need to use this for #718

**Pull request merged**